controllers/machinehealthcheck_controller.go

```diff
@@ -351,7 +355,8 @@ func (r *MachineHealthCheckReconciler) patchHealthyTargets(ctx context.Context,
 				}
 			}
 		}
-
+		// mark machine in healthy status after delete externalRemediation object
+		conditions.MarkTrue(t.Machine, clusterv1.MachineHealthCheckSuccededCondition)
```

Comment:
Moving the condition set here seems like a good improvements from a quick look, given that `healthCheckTargets` should have been read only from what I can gather from the original code comments.

cc @JoelSpeed

Reply:
```suggestion
		// Mark machine in healthy status if it's healthy.
		conditions.MarkTrue(t.Machine, clusterv1.MachineHealthCheckSuccededCondition)
```

Not every Machine has a RemediationTemplate attached to it

Reply:
Whether there is no RemediationTemplate or not, as long as the node is judged to be healthy, the healthy condition on the machine patch shall be provided

Reply:
Right, suggestions is to update comment to reflect that.